app/src/main/java/com/mailerdaemon/app/impContacts/ContactDetailFragment.kt

```diff
@@ -0,0 +1,126 @@
+package com.mailerdaemon.app.impContacts
+
+import android.annotation.SuppressLint
+import android.app.SearchManager
+import android.content.Context
+import android.content.Intent
+import android.net.Uri
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.Menu
+import android.view.MenuInflater
+import android.view.View
+import android.view.ViewGroup
+import android.view.inputmethod.InputMethodManager
+import androidx.appcompat.widget.SearchView
+import androidx.fragment.app.Fragment
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.google.gson.Gson
+import com.mailerdaemon.app.R
+import com.mailerdaemon.app.toast
+import com.mailerdaemon.app.utils.ContactFunction
+import kotlinx.android.synthetic.main.fragment_contact_detail.*
+import kotlinx.android.synthetic.main.fragment_contact_detail.view.*
+import java.io.IOException
+import java.nio.charset.StandardCharsets
+import java.util.ArrayList
+
+class ContactDetailFragment : Fragment(), ContactFunction {
+    private var contactList = ArrayList<Contact>()
+    private val adapter: ContactRecyclerAdapter = ContactRecyclerAdapter(this)
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        contactList = Gson().fromJson(loadJSONFromAsset(arguments
+                ?.getString("type")), FacultyModel::class.java)
+                .contact as ArrayList<Contact>
```

Comment:
`as` is unsafe cast. Check if the result is proper else assign an empty list. Use safe nullable cast `as?`.